Dockerfile

```diff
@@ -1,4 +1,24 @@
-FROM golang:1.14-buster
+FROM python:3.8.5-buster as code
+
+WORKDIR /root/aladdin
+
+RUN python -m venv /root/.venv
+ENV PATH /root/.venv/bin:$PATH
+
+# On some images "sh" is aliased to "dash" which does not support "set -o pipefail".
+# We use the "exec" form of RUN to delegate this command to bash instead.
+# This is all because we have a pipe in this command and we wish to fail the build
+# if any command in the pipeline fails.
+ARG POETRY_VERSION=1.0.10
+RUN ["/bin/bash", "-c", "set -o pipefail && curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python"]
+ENV PATH /root/.poetry/bin:$PATH
+
+ARG POETRY_VIRTUALENVS_CREATE="false"
+# Install aladdin python requirements
+COPY pyproject.toml poetry.lock ./
+RUN poetry install --no-root
```

Comment:
What is gained by running just this in a separate image?  Are we avoiding garbage left in the filesystem somewhere by installing here and then copying over .venv?

Reply:
the total aladdin image gets to be way smaller! we need to use debian-buster for installation of requirements but then drop down to debian-buster-slim for the final image